reference/promise-types/files.markdown

```diff
@@ -319,34 +319,57 @@ Form of the permissions is as follows:
                         };
 ```
 
-*   `user`
+* `user`
 
-indicates that the line applies to a user specified by the user identifier
-`uid`. `mode` is the permission mode string.
+A valid username identifier for the system and cannot be empty. However, `user`
+can be set to `*` as a synonym for the entity that owns the file system object
+(e.g. ```user:*:r```).
 
-*   `group`
-
-
-Indicates that the line applies to a group specified by the group identifier
-`gid`. `mode` The permission mode string.
-
-* `all`
+**Notes:**
 
-Indicates that the line applies to every user. `mode` is the permission mode
-string.
+** The users id is not a valid alternative.
+** This ACL is **required** when `acl_method` is set to `overwrite`
 
 * `uid`
 
 A valid user identifier for the system and cannot be empty. However, `uid` can
 be set to `*` as a synonym for the entity that owns the file system object
-(e.g. user:\*:r).
+(e.g. `uid:*:r`).
+
+**Note:** The username is not a valid alternative.
+
+* `group`
+
+A valid group identifier for the system and cannot be empty. However, `group`
+can be set to `*` as a synonym for the group that owns the POSIX file system
+object (```group:*:rwx```).
+
+**Notes:**
+
+** The groups id is not a valid alternative.
+** This ACL is **required** when `acl_method` is set to `overwrite`.
 
 * `gid`
 
 A valid group identifier for the system and cannot be empty. However, in some
 ACL types, `gid` can be set to `*` to indicate a special group (e.g. in POSIX
 this refers to the file group).
 
+**Note:** The group name is not a valid alternative.
```

Comment:
I think group name is fine. I'm not sure that gid works.

Reply:
oh, sorry, gid should be an id, group should be a group-name. got it.

Reply:
OK, feel free to suggest different wording that mightmake it more clear